scripts/stablex/utilities.js

```diff
@@ -143,14 +167,88 @@ const sendLiquidityOrders = async function(
   )
 }
 
+async function mintOWL({ account, amount, owl }) {
+  console.log("Mint %d of OWL for user %s", amount, account)
+  return owl.mintOWL(account, amount)
+}
+
+async function deleteOrders({ orderIds, accounts, batchExchange }) {
+  console.log("Cancel %d orders for %d users", orderIds.length, accounts.length)
+  for (let i = 0; i < orderIds.length; i++) {
+    const orderId = orderIds[i]
+    const account = accounts[i]
+    const cancelReceipt = await batchExchange.cancelOrders([orderId], { from: account })
+    const events = cancelReceipt.logs.map(log => log.event).join(", ")
+    console.log("Canceled/Deleted order %s for user %s. Emitted events: %s", orderId.toString(10), account, events)
+  }
+}
+
+async function getBatchId(batchExchange) {
+  const batchId = await batchExchange.getCurrentBatchId.call()
+  return batchId.toNumber()
+}
+
+async function submitSolution({ name, batchId, solution, solverAddress, batchExchange }) {
+  console.log(`Submit "${name}":
+  - Objective value: ${solution.objectiveValue}
+  - Touched orders: ${solution.touchedorderIds.join(", ")}
+  - Volumes: ${solution.volumes.join(", ")}
+  - Prices: ${solution.prices.join(", ")}
+  - Token ids for prices: ${solution.tokenIdsForPrice.join(", ")}`)
+  const objectiveValue = await batchExchange.submitSolution(
+    batchId,
+    1,
+    solution.owners,
+    solution.touchedorderIds,
+    solution.volumes,
+    solution.prices,
+    solution.tokenIdsForPrice,
+    { from: solverAddress }
+  )
+  console.log(`Transaction for ${name}: ${objectiveValue.tx}`)
+}
+
+async function setAllowances({ users, amount, batchExchange, tokens }) {
+  for (let i = 0; i < users.length; i++) {
+    for (let j = 0; j < tokens.length; j++) {
+      await setAllowance({
+        token: tokens[j],
+        account: users[i],
+        amount,
+        batchExchange,
+      })
+    }
+  }
+}
+
+async function setAllowance({ token, account, amount, batchExchange }) {
+  console.log("Set allowance of %d for token %s and user %s", amount, token.address, account)
+  await token.approve(batchExchange.address, amount, { from: account })
+}
+
+async function mintOwlForUsers({ users, amount, owl }) {
+  for (let i = 0; i < users.length; i++) {
+    await mintOWL({ account: users[i], amount, owl })
+  }
+}
+
 const maxUint32 = new BN(2).pow(new BN(32)).sub(new BN(1))
 
 module.exports = {
+  getOwl,
+  getBatchExchange,
   addTokens,
   closeAuction,
   token_list_url,
   fetchTokenInfo,
   sendLiquidityOrders,
   getOrdersPaginated,
   maxUint32,
+  setAllowance,
+  setAllowances,
+  mintOwlForUsers,
+  mintOWL,
```

Comment:
Instead of exporting the singular and plural methods, would it make sense to just use the plural with a one element list? The more methods you export the more confusing it becomes.

Reply:
okay, no problem. Removing singulars

Reply:
47016fbf956b13bbbae3e5d98dc4bc78404b6e0d